wagtail/models/__init__.py

```diff
@@ -177,7 +177,11 @@ def get_streamfield_names(model_class):
 
 class BasePageManager(models.Manager):
     def get_queryset(self):
-        return self._queryset_class(self.model).order_by("path")
+        qs = self._queryset_class(self.model).order_by("path")
+        # Restrict proxy model querysets to the exact type
+        if self.model._meta.proxy:
```

Comment:
suggestion: @ababic, given the conversation below ([the one where @chosak asked about the change to exact type](https://github.com/wagtail/wagtail/pull/9255/files#r980373046)), perhaps a longer developer comment is needed. I suspect folks will be asking in #support, so would be nice to be able point them to a bit of docs

Reply:
👍

Reply:
Resolved in 1b1cb521d0